rosetta-lang/src/main/java/com/regnosys/rosetta/validation/RosettaSimpleValidator.xtend

```diff
@@ -1173,11 +1174,15 @@ class RosettaSimpleValidator extends AbstractDeclarativeRosettaValidator {
 				return
 			}
 			val segments = container.path.asSegmentList(container.path)
-			val attr = segments?.last?.attribute
-			if (!attr.hasReferenceAnnotation) {
-				error(''''«o.operator»' can only be used with attributes annotated with [metadata reference] annotation.''',
-					o, ROSETTA_OPERATION__OPERATOR)
+			val feature = segments?.last?.feature
+			if (feature instanceof Attribute) {
+				if (!feature.hasReferenceAnnotation) {
```

Comment:
If feature is a `MetaType`, this error will not be triggered, but it should be. `as-key` should not be useable when setting a `scheme`.